ledger/participant-integration-api/src/main/scala/platform/store/dao/events/contracts/TranslationCacheBackedContractsStore.scala

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events.contracts
+
+import java.time.Instant
+
+import com.daml.ledger.participant.state.index.v2.ContractStore
+import com.daml.ledger.resources.Resource
+import com.daml.lf.data.Ref
+import com.daml.lf.transaction.GlobalKey
+import com.daml.logging.LoggingContext
+import com.daml.platform.store.dao.events._
+
+import scala.concurrent.Future
+
+class TranslationCacheBackedContractsStore(
```

Comment:
I am not sure whether this is really necessary now, or part of asthetics...
I would like to emphasize that now everything done below ...store.dao will have to be done twice.
Would it be possible not doing refactorings, make minimal changes to LedgerDao, and add implementation only to append-only?